internal/crypto/crypto.go

```diff
@@ -77,6 +78,41 @@ func GenerateEmergencySSHCAKey(seed []byte) (ssh.Signer, error) {
 	return ca, nil
 }
 
+// GenerateSignedSSHHostKey creates a host key that is signed by the given CA.
+func GenerateSignedSSHHostKey(principals []string, ca ssh.Signer) (*pem.Block, *ssh.Certificate, error) {
+	hostKeyPub, hostKey, err := ed25519.GenerateKey(nil)
```

Comment:
Why do we generate the key here, instead of receiving a public key and signing that?